app/models/local_authority.rb

```diff
@@ -7,9 +7,9 @@ class LocalAuthority < ApplicationRecord
       message: "%{value} is not a valid tier",
     }
 
-  has_many :links
-  belongs_to :parent_local_authority, foreign_key: :parent_local_authority_id, class_name: "LocalAuthority"
-  has_many :service_tiers, foreign_key: :tier_id, primary_key: :tier_id
+  has_many :links, dependent: :destroy
```

Comment:
I wonder if we should just use `restrict_with_error` everywhere, to be on the safe side?

Reply:
There are some I'm quite confident with e.g. in this case, a Local Authority has many links, and if the Local Authority no longer exists there's no point in keeping its links.  However I'll do another check and change anything where it's less clear-cut.